cpu/cortexm_common/cortexm_init.c

```diff
@@ -0,0 +1,52 @@
+/*
+ * Copyright (C) 2015 Freie Universität Berlin
+ *
+ * This file is subject to the terms and conditions of the GNU Lesser
+ * General Public License v2.1. See the file LICENSE in the top level
+ * directory for more details.
+ */
+
+/**
+ * @ingroup     cpu_cortexm_common
+ * @{
+ *
+ * @file
+ * @brief       Cortex-M specific configuration and initialization options
+ *
+ * @author      Hauke Petersen <hauke.petersen@fu-berlin.de>
+ *
+ * @}
+ */
+
+#include "cpu.h"
+
+/**
+ * @name   Pattern to write into the co-processor Access Control Register to
+ *         allow full FPU access
+ */
+#define FULL_FPU_ACCESS         (0x00f00000)
+
+void cortexm_init(void)
+{
+    /* initialize the FPU on Cortex-M4F CPUs */
+#ifdef CPU_ARCH_CORTEX_M4F
+    /* give full access to the FPU */
+    SCB->CPACR |= (uint32_t)FULL_FPU_ACCESS;
```

Comment:
done: #3112

Regarding the FPU initialization above: I makes much sense to leave it in. With #3112 the FPU should not be used. But if it is under some unforeseeable circumstances is used (though I don't know how this could happen), this initialization would prevent the CPU from going into hard-fault.